File: languages/go/entity/key/local_key.go
package key

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// LocalKey represents a local key with its public and private components.
type LocalKey struct {
	// Key is the public key.
	Key        string
	// PrivateKey is the private key.
	PrivateKey string
	// KeyType is the type of the key.
	KeyType    KeyType
}

func NewLocalKeyFromProto(s *proto.LocalKey) LocalKey {
	if s == nil {
		return LocalKey{}
	}
	return LocalKey{
		Key:        s.GetKey(),
		PrivateKey: s.GetPrivateKey(),
		KeyType:    KeyTypeFromProto[s.KeyType],
	}
}

func (s LocalKey) ToProto() *proto.LocalKey {
	return &proto.LocalKey{
		Key:        s.Key,
		PrivateKey: &s.PrivateKey,
		KeyType:    KeyTypeToProto[s.KeyType],
	}
}

File: languages/js/src/entity/key/local_key.ts
import * as keysEntitiesProto from "../../bridge/proto/keys_entities";
import { KeyType } from "./key_type";

/**
 * Represents a local key with its public and private components.
 */
export class LocalKey {
  /**
   * Is the public key.
   */
  public key: string;
  /**
   * Is the private key.
   */
  public privateKey?: string;
  /**
   * Is the type of the key.
   */
  public keyType: KeyType;

  /**
   * Constructs a LocalKey object with the specified parameters.
   * @param key 
   * @param keyType 
   * @param privateKey 
   */
  constructor(key: string, keyType: KeyType, privateKey?: string) {
    this.key = key;
    this.privateKey = privateKey;
    this.keyType = keyType;
  }

  public toProto(): keysEntitiesProto.LocalKey {
    return keysEntitiesProto.LocalKey.fromPartial({
      key: this.key,
      privateKey: this.privateKey,
      keyType: KeyType.toProto(this.keyType)
    });
  }

  static fromProto(r: keysEntitiesProto.LocalKey): LocalKey {
    return new LocalKey(r.key, KeyType.fromProto(r.keyType), r.privateKey);
  }
}
